Fix conflict with Laravel/Scout v10.11.6 and newer

mariuscucuruz opened the pull request:
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no     
| Related Issue     | Fix #347, [#883](https://github.com/laravel/scout/issues/883), [#855](https://github.com/laravel/scout/issues/855)
| Need Doc update   | no


## Describe your change

Downgraded `laravel/scout` to version v10.11.5 in composer.json to resolve a conflict with the Algolia Search Client.

### Details 

This change addresses compatibility issues between [Laravel Scout v10.11.6 (and newer)](https://packagist.org/packages/laravel/scout#v10.11.6) and recent versions Algolia Search Client, ensuring stable integration.

The proposed solution is to downgrade downgrade dependency [Laravel Scout to v10.11.5](https://packagist.org/packages/laravel/scout#v10.11.5).

### Related Issues

- [Incompatible with `laravel/scout` v10.11.6 #347](https://github.com/algolia/scout-extended/issues/347)
- [Abstract method `performSearch` and `deleteIndex` are not implemented #883](https://github.com/laravel/scout/issues/883)
- [Algolia Search PHP Client v4 is not Compatible #855](https://github.com/laravel/scout/issues/855)

### Context

Laravel Scout [v10.11.6](https://packagist.org/packages/laravel/scout#v10.11.6) and [v10.11.7](https://packagist.org/packages/laravel/scout#v10.11.7) have conflicts with all recent versions of [Algolia Search Client](https://packagist.org/packages/algolia/algoliasearch-client-php).

```
    "conflict": {
        "algolia/algoliasearch-client-php": "<3.2.0|>=5.0.0"
    },

```

DevinCodes commented on composer.json:
```diff
@@ -30,7 +30,7 @@
         "illuminate/filesystem": "^9.0|^10.0|^11.0",
         "illuminate/support": "^9.0|^10.0|^11.0",
-        "laravel/scout": "^9.0|^10.0",
+        "laravel/scout": "^9.0|^10.0|<10.11.7",
```
```suggestion
        "laravel/scout": "^9.0|^10.0 <=10.11.5",
```

mariuscucuruz replied:
fair enough! thanks

DevinCodes reviewed (changes requested):
Hi @mariuscucuruz ,

Thank you for the PR. A small change is needed in the version constraint it seems, as the CI is still pulling in the version we're incompatible with (since that matches `^10.0`).

In order for the CI to pass, you'll also need to remove line 10 in the `phpstan.neon.dist` file, which seems odd since that line was needed before to have the CI pass. You can do this manually, or cherry-pick commit f319b644dff9595888eb017c24d0737803879e32 onto your branch, whichever works best for you 🙂 

Thank you in advance, I'll merge & deploy this once the changes have been applied and the CI passes!

mariuscucuruz commented:
> 
> Thank you for the PR. A small change is needed in the version constraint it seems, as the CI is still pulling in the version we're incompatible with (since that matches `^10.0`).
> 
> In order for the CI to pass, you'll also need to remove line 10 in the `phpstan.neon.dist` file, which seems odd since that line was needed before to have the CI pass. You can do this manually, or cherry-pick commit [f319b64](https://github.com/algolia/scout-extended/commit/f319b644dff9595888eb017c24d0737803879e32) onto your branch, whichever works best for you 🙂
> 

Hey @DevinCodes,

I've updated composer to `"laravel/scout": "^9.0||^10.0 <10.11.6",` to prevent limit Scout to v10.11.5.
I've also removed the line as per [commit f319b644](https://github.com/algolia/scout-extended/commit/f319b644dff9595888eb017c24d0737803879e32). 

However the CI is still failing for some reason. What would you suggest to fix this?

Cheers.

DevinCodes commented:
Thank you for the updates @mariuscucuruz ! The CI was failing due to permission setup, I just triggered it again and the tests are passing!

I'll merge and release this later today!

DevinCodes commented:
This has been released in [v3.1.1](https://packagist.org/packages/algolia/scout-extended#v3.1.1). Thank you again for the contribution!
